roblox/bases/basecatalogitem.py

```diff
@@ -0,0 +1,47 @@
+"""
+
+This file contains the BaseCatalogItem object, which represents a Roblox catalog item ID.
+
+"""
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+from .baseitem import BaseItem
+
+if TYPE_CHECKING:
+    from ..client import Client
+
+
+class BaseCatalogItem(BaseItem):
```

Comment:
catalog items are either bundles or assets. This approach of one unified class makes it impossible to invoke methods that BaseAsset or a hypothetical BaseBundle would support even when applicable. Maybe forego this class entirely, introduce a BaseBundle (which we need anyway), and make "catalog items" just a `BaseAsset | BaseBundle` union? Or, instead of a union, both could derive from a new BaseCatalogItem class with a `@property` that indicates what "type" of catalog item it is (asset or bundle) determined by `isinstance(self, BaseAsset): "asset"` etc. not sure.